lib/tbot/renew.go

```diff
@@ -293,6 +293,73 @@ func (b *Bot) getRouteToDatabase(ctx context.Context, client auth.ClientI, dbCfg
 	}, nil
 }
 
+func getApp(ctx context.Context, client auth.ClientI, appName string) (types.Application, error) {
+	res, err := client.ListResources(ctx, proto.ListResourcesRequest{
+		Namespace:           defaults.Namespace,
+		ResourceType:        types.KindAppServer,
+		PredicateExpression: fmt.Sprintf(`name == "%s"`, appName),
+		Limit:               int32(defaults.DefaultChunkSize),
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	servers, err := types.ResourcesWithLabels(res.Resources).AsAppServers()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var apps []types.Application
+	for _, server := range servers {
+		apps = append(apps, server.GetApp())
+	}
+	apps = types.DeduplicateApps(apps)
+
+	// Check app name in the (unlikely) event the server does not process
+	// filters.
+	for _, app := range apps {
```

Comment:
How old of a Teleport server version are we expecting to not support these filters ? It would certainly be simpler if we just called ListResources with a limit of one.

Reply:
This seems to be yet another case where our deduplication makes some assumptions that could bite us. Should we at least log a warning if we got more than one app with the same name?

Reply:
I think given we expect this to run against only versions that do support the predicate, we should look to drop this additional check. I'm not sure I like to start setting a precedent of clients getting into the habit of double-guessing results produced by the server, unless we have some historical context I'm missing that points to this often being an issue?

Reply:
I've simplified the check. Looks like I even did this already for databases, if only I'd scrolled up 30 lines.